File: ARCADE/The Core/Level 3_4: mirrorBits.cs
    using System;
     
    public class Test
    {
    	public static int mirrorBits(int a) {
            int rs = 0;
            while(a > 0){
                rs = rs * 2 + (a % 2);
                a /= 2;
            }
            return rs;
        }
        
    	public static void Main()
    	{
    		int a = 97;
    		Console.Write(mirrorBits(a));
    	}
    }

File: ARCADE/The Core/Level 3_4: mirrorBits.pas
program mirrorBits;

function mirrorBits(a: integer): integer;
var rs : integer;   
begin
    rs := 0;
    while(a > 0) do
    begin
        rs := rs * 2 + (a mod 2);
        a := a div 2;
    end;
    mirrorBits:= rs;
end;

var a : integer;
begin
	a := 97;
	writeln(mirrorBits(a));
	a := 8;
	writeln(mirrorBits(a));
end.
